### PR TITLE
[FIX] throw warning a masker `generate_report` is called without matplotlib

### LINKED ISSUE #4456: [BUG] inconsistent masker behavior  for `generate_report` when matplolib is missing
Note that there is inconsistent behavior amongst the other maskers in nilearn regarding their `generate_report` when matplolib is missing.

NiftiMasker throws an error

```python
>>> from nilearn.maskers import NiftiMasker
>>> masker = NiftiMasker()
>>> masker.generate_report()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/remi/github/nilearn/nilearn/nilearn/maskers/nifti_masker.py", line 310, in generate_report
    from nilearn.reporting.html_report import generate_report
  File "/home/remi/github/nilearn/nilearn/nilearn/reporting/__init__.py", line 9, in <module>
    from nilearn.reporting.glm_reporter import make_glm_report
  File "/home/remi/github/nilearn/nilearn/nilearn/reporting/glm_reporter.py", line 22, in <module>
    from matplotlib import pyplot as plt
ModuleNotFoundError: No module named 'matplotlib'
```

NiftiSpheresMasker throws an warning

```python
>>> from nilearn.maskers import NiftiSpheresMasker
>>> masker = NiftiSpheresMasker(seeds=(0,0,0))
>>> masker.generate_report()
/home/remi/github/nilearn/nilearn/nilearn/maskers/nifti_spheres_masker.py:409: ImportWarning: Matplotlib is not imported! No reports will be generated.
  warnings.warn(category=ImportWarning, message=mpl_unavail_msg)
[None]
```

Noticed this as the correct warnings in the maskers code are not tested.

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/pull/4432#discussion_r1646348441_

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4456

Changes proposed in this pull request:
- move where warning is actually thrown
- add tests for the behavior

based on #4432 so needs to be merged after it (requires some conftest refactoring)

### CODE REVIEW

**Comment by bthirion** [comment]
Should be ready for review soon ?

**Comment by Remi-Gau** [comment]
> Should be ready for review soon ?

Indeed: ready for review now

File: nilearn/maskers/tests/test_nifti_maps_masker.py [hunk 1]
```diff
@@ -573,3 +574,24 @@ def test_3d_images():
     assert epis.shape == (1, 3)
     epis = masker.fit_transform([epi_img1, epi_img2])
     assert epis.shape == (2, 3)
+
+
+@pytest.mark.skipif(
+    have_mpl, reason="Test requires matplotlib to be not installed."
```
Comment (bthirion) [inline nilearn/maskers/tests/test_nifti_maps_masker.py:580]: I guess that the canonical warning would be "Test requires matplotlib not to be  installed."

**Review by bthirion — COMMENTED** [review]
LGTM.

File: nilearn/maskers/tests/test_nifti_maps_masker.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/maskers/tests/test_nifti_maps_masker.py:580]: Will fix this one and merge

**Pull request merged**